app/scripts/views/components/virtualStudy/virtualStudy.js

```diff
@@ -0,0 +1,353 @@
+                              'study?id=' + self_.savedVC.id + '" onclick="window.open(\'' +
+                              window.cbioURL + 'study?id=' + self_.savedVC.id + '\')">' +
+                              window.cbioURL + 'study?id=' + self_.savedVC.id + '</a>');
+
+                            self_.hideLoading(tooltip);
+                            self_.showShared(tooltip);
+                            previousSelectedCases = currentSelectedCases;
+                          })
+                          .fail(function() {
+                            self_.hideLoading(tooltip);
+                            self_.hideDialog(tooltip);
+                            self_.updateFailedMessage(tooltip,
+                              '<i class="fa fa-exclamation-triangle"></i>' +
+                              '<span class="left-space">' +
+                              'Failed to share virtual study, ' +
+                              'please try again later.</span>')
+                            self_.showFailedInfo(tooltip);
+                          });
+                      } else {
+                        // Hide saving icon if current study is same as previous.
+                        tooltip.find('.saving').css('display', 'none');
+                      }
+                    }
+                  });
+                }
+              });
+              tooltip.find('.copy-cohort-btn').click(function() {
+                self_.hideDialog(tooltip);
+                self_.hideShared(tooltip);
+                self_.hideLoading(tooltip);
+
+                api.reposition();
+
+                // Copy virtual study link to clipboard
+                var temp = $("<input>");
+                $("body").append(temp);
+                temp.val(tooltip.find('.virtual-study-link').attr('href')).select();
+                // execCommand('copy') allows to run commands to copy the contents of selected editable region.
+                document.execCommand("copy");
+                // Check if users copy url successfully
+                if (temp.val() === tooltip.find('.virtual-study-link').attr('href')) {
+                  self_.hideShared(tooltip);
+                  self_.showAfterClipboard(tooltip);
+                  api.reposition();
+                }
+                temp.remove();
+              });
+              tooltip.find('.cohort-name')
+                .keyup(function() {
+                  self_.checkVSName(tooltip);
+                });
+              this.createdQtip = true;
+            },
+            show: function() {
+              var tooltip = $('.iviz-virtual-study-btn-qtip .qtip-content');
+              self_.updateStats = true;
+              self_.$nextTick(function() {
+                // If user hasn't specific name only.
+                if (tooltip.find('.cohort-name').val() === '') {
+                  tooltip.find('.cohort-name').val(self_.virtualStudyName);
```

Comment:
May be just a placeholder when nothing is set?

Reply:
I thought we agreed to use the default name as always?

Reply:
true, but if user want's to change the name, he needs to delete the existing name and enter new one. instead check it at when the save button is clicked.

Reply:
I'm just afraid the placeholder is not obvious enough letting them know this will be saved if no named specified. Again, we shouldn't spend too much time discussing the details about this. Everything will change after sending out the testing website.